### PR TITLE
Updating changelogs for 4.2 release

### PR DESCRIPTION
**Pull request by aaronjeline**
## Description of changes
Updates changelogs
## Issue #, if available
N/A
## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [X] A change (breaking or otherwise) that only impacts unreleased or experimental code.

I confirm that this PR (choose one, and delete the other options):

- [X] Updates the "Unreleased" section of the CHANGELOG with a description of my change (required for major/minor version bumps).

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [X] Does not require updates because my change does not impact the Cedar formal model or DRT infrastructure.

I confirm that [`docs.cedarpolicy.com`](https://docs.cedarpolicy.com/) (choose one, and delete the other options):

- [X] Does not require updates because my change does not impact the Cedar language specification.

### CODE REVIEW

**Review by cdisselkoen — APPROVED** [review]

File: cedar-policy/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ Starting with version 3.2.4, changes marked with a star (*) are _language breaki
 ## [Unreleased]
 Cedar Language Version: TBD
 
+## [4.2.0] - 2024-10-07
```
Comment (cdisselkoen) [inline cedar-policy/CHANGELOG.md:15]:
```suggestion
## [4.2.0] - 2024-10-07
Cedar Language version: 4.1
```

**Review by adpaco-aws — APPROVED** [review]

**Review by john-h-kastner-aws — CHANGES_REQUESTED** [review]
Where's #1258 in the change log

File: cedar-policy/CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +24,8 @@ Cedar Language Version: TBD
 - Stabilized [RFC 82](https://github.com/cedar-policy/rfcs/pull/82), removing
   the experimental `entity-tags` feature flag. That functionality is now available
   without the feature flag.
+- Improved validation error messages around action ids (#1258, resolving
```
Comment (john-h-kastner-aws) [inline cedar-policy/CHANGELOG.md:27]: I see it's actually listed under 4.1.0

Comment (adpaco-aws) [reply cedar-policy/CHANGELOG.md]: Good catch, didn't notice I had put it under 4.1... my fault.

@aaronjeline do you mind moving it?

Comment (aaronjeline) [reply cedar-policy/CHANGELOG.md]: yup

Comment (john-h-kastner-aws) [reply cedar-policy/CHANGELOG.md]: looks like it was a funny merge. I looked at your PR and it's in the right place.

**Review by john-h-kastner-aws — APPROVED** [review]

File: cedar-policy/CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,9 @@ Starting with version 3.2.4, changes marked with a star (*) are _language breaki
 ## [Unreleased]
 Cedar Language Version: TBD
 
+## [4.2.0] - 2024-10-07
```
Comment (john-h-kastner-aws) [inline cedar-policy/CHANGELOG.md:15]: Needs the link target at the bottom

**Review by john-h-kastner-aws — APPROVED** [review]

**Pull request merged**